codeGenerator/src/robotConfiguration/robotConfig.cs

```diff
@@ -92,23 +92,51 @@ robotVariants loadRobotConfiguration(string fullPathName)
                 }
             }
 
-            foreach (robot theRobot in theRobotVariants.robot)
+            //try loading any additional mechanisms in cofiguration directory
+            string[] files = Directory.GetFiles(Path.GetDirectoryName(fullPathName), "*.xml");
+            foreach (string file in files)
             {
-                foreach (mechanismInstance mi in theRobot.mechanismInstance)
+                if (theRobotVariants.mechanism.Any(p => p.name == Path.GetFileName(file).Replace(".xml", "")))
                 {
-                    string mechanismFullPath = Path.Combine(Path.GetDirectoryName(fullPathName), mi.mechanism.name + ".xml");
+                    //if we have previously loaded the mechanism, don't load it again
+                    continue;
+                }
+                else
+                {
+                    string mechanismFullPath = Path.Combine(Path.GetDirectoryName(fullPathName), file);
+
+                    string tempFile = File.ReadAllText(mechanismFullPath);
 
-                    //addProgress("Loading mechanism configuration " + mechanismFullPath);
-                    mySerializer = new XmlSerializer(typeof(mechanism));
                     using (var myFileStream = new FileStream(mechanismFullPath, FileMode.Open))
                     {
-                        mechanism m = (mechanism)mySerializer.Deserialize(myFileStream);
+                        //ignore configuration files
+                        if(!tempFile.Contains("robotVariants") && !tempFile.Contains("toolConfiguration"))
```

Comment:
Why not place 

if(!tempFile.Contains("robotVariants") && !tempFile.Contains("toolConfiguration"))

outside of the using statement?